vegetation/patch/space.py

```diff
@@ -73,21 +74,23 @@ def __init__(self, bounds, epsg, model):
         # the bounds of the study area, so that we can grab if we already have it
         self.bounds_md5 = hashlib.md5(str(bounds).encode()).hexdigest()
 
-        self.pystac_client = None
-        if not LOCAL_STAC_CACHE_FSTRING:
-            self.pystac_client = PystacClient.open(
-                DEM_STAC_PATH, modifier=planetary_computer.sign_inplace
-            )
-
-    def get_elevation(self):
+    @cached_property
+    def pystac_client(self):
+        return PystacClient.open(
+            DEM_STAC_PATH, modifier=planetary_computer.sign_inplace
+        )
```

Comment:
Now the client will be made available on-demand, and cached after it's been initialized.